lib/common/digest.c

```diff
@@ -250,28 +250,35 @@ pcmk__xa_filterable(const char *name)
 char *
 crm_md5sum(const char *buffer)
 {
-    int lpc = 0, len = 0;
+    unsigned int dlen = gnutls_hash_get_len(GNUTLS_DIG_MD5);
+    unsigned char *raw_digest = NULL;
     char *digest = NULL;
-    unsigned char raw_digest[MD5_DIGEST_SIZE];
+    int rc = 0;
 
-    if (buffer == NULL) {
-        buffer = "";
+    if (dlen == 0) {
+        return NULL;
```

Comment:
handle NULL argument too (returning NULL is fine since we ensure non-NULL internally)

Reply:
Fixed.